src/diffusers/models/unet_2d_condition.py

```diff
@@ -123,20 +124,45 @@ def __init__(
         num_class_embeds: Optional[int] = None,
         upcast_attention: bool = False,
         resnet_time_scale_shift: str = "default",
+        time_embed_dim: int = None,
+        time_embedding_type: str = "positional", # fourier, positional
+        timestep_act_2: bool = False, 
+        class_labels_dim: Optional[int] = None,
+        conv_in_kernel: int = 3,
+        conv_out_kernel: int = 3,
+        norm_out: bool = True,
+        attn1_types: Tuple[bool, None]=(None,'self','self','self'),
+        attn2_types: Tuple[bool, None] = (None, 'cross','cross','cross'),
+        downsample: Optional[Tuple[bool]] = None,
+        upsample: Optional[Tuple[bool]] = None,
+        skip_freq: str = "layer" # layer, block
```

Comment:
Would be nice to remove this argument and instead automatically handle this in the new up-/downsampling blocks